lib/service/task_service.dart

```diff
@@ -1,19 +1,21 @@
 import 'package:task_list_app/client/backend_client.dart';
+import 'package:task_list_app/models/task.dart';
 
 class TaskService {
   const TaskService();
   static const String taskApiBase = '/tasks';
 
-  Future<void> addTask({required String taskName}) async {}
+  Future<void> addTask({required Task taskName}) async {}
 
-  Future<String> readTasks() async {
+  Future<List<Task>> readTasks() async {
     BackendClient client = const BackendClient();
-    return await client.getData(uri: taskApiBase);
+    List<Map<String, String>> apiTask = await client.getData(uri: taskApiBase);
+    return apiTask.map((task) => Task.fromJson(task)).toList();
```

Comment:
I would wrap line 13 in a `try`/`catch` block in case the response from the back-end is not at all what you were expecting (i.e. an error from Spring or something). For the catch, you can simply `print()` at this point.

Reply:
Can you explain this again in the context of the changes I made?

Reply:
After the changes, I think what I said applies to `backend_client.dart`. Wrapping the API call and the JSON decode in a `try` block will help if anything goes wrong on those two lines. In the `catch`, you can just `print('Error: $e');` or something.

Reply:
```
  Future<dynamic> getData({required String uri}) async {
    var url = Uri.http(localhost, uri);
    try {
      var response = await http.get(url);
      dynamic data = jsonDecode(response.body);

      if (response.statusCode == 200) {
        return data;
      } else {
        return 'HTTP Request failed with status: ${response.statusCode}';
      }
    } catch (e) {
      print('Error during HTTPrequest $e');
      return 'Error during HTTP request: $e';
    }
  }
  ```
  
  This is what I have on my end for the `backend_client.dart`. Does that work?

Reply:
I should say, that's what I already had. Did you mean something different?

Reply:
Oh my apologies. I am not reviewing carefully apparently. Looks great.